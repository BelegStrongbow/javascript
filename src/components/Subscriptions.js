import React from "react";
import Subscription from "./Subscription";
import { Table, Zebra } from "./Tables";
import Paper from "./Paper";

/**
 *
 * @param {Object} props The props to use
 *
 * @returns {ReactElement} The rendered component.
 */
export default function Subscriptions( props ) {
	return (
		<Table>
			<Paper>
				<Zebra>
					{ props.subscriptions.map( function( subscription ) {
						let onManageHandler = () => {
							props.onManage( subscription.id );
						};
						return <Subscription
							key={ subscription.id }
							id={ subscription.id }
							icon={ subscription.icon }
							name={ subscription.name }
							used={ subscription.used }
							limit={ subscription.limit }
							nextPayment={ subscription.nextPayment }
							billingAmount={ subscription.billingAmount }
							billingCurrency={ subscription.billingCurrency }
							onManage={ onManageHandler }
						/>;
					} )
					}
				</Zebra>
			</Paper>
		</Table>
	);
}

Subscriptions.propTypes = {
	subscriptions: React.PropTypes.arrayOf(
		React.PropTypes.shape(
			{
				id: React.PropTypes.string.isRequired,
				icon: React.PropTypes.string.isRequired,
				name: React.PropTypes.string.isRequired,
				used: React.PropTypes.number.isRequired,
				limit: React.PropTypes.number.isRequired,
				nextPayment: React.PropTypes.instanceOf( Date ).isRequired,
				billingAmount: React.PropTypes.number.isRequired,
				billingCurrency: React.PropTypes.string.isRequired,
			}
		)
	),
	onManage: React.PropTypes.func.isRequired,
};
