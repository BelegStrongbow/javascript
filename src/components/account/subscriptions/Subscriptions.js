/* eslint-disable */
import PropTypes from "prop-types";
import React from "react";
import SubscriptionRow from "./SubscriptionRow";
import { ListTable } from "../../Tables";
import { Paper } from "../../PaperStyles";
import { injectIntl } from "react-intl";
import forEach from "lodash/forEach";

/**
 *
 * @param {Object} props The props to use
 *
 * @returns {ReactElement} The rendered component.
 */
class Subscriptions extends React.Component {
	outputSubscriptionRows() {
		const subscriptionRows = [];
		forEach( this.props.subscriptions, ( subscriptionsGroupedByProduct ) => {
			subscriptionRows.push( subscriptionsGroupedByProduct );
		} );

		return subscriptionRows;
	}

	render() {
		return (
			<Paper>
				<ListTable>
					{
						this.outputSubscriptionRows()
							.map( ( subscriptionsArray ) => {
								return <SubscriptionRow
									key={ subscriptionsArray[ 0 ].id }
									subscriptionsArray={ subscriptionsArray }
									onManage={ this.props.onManage }
									isGrouped={ this.props.isGrouped }
									needsAttention={ this.props.needsAttention }
									showDetailsModal={ this.props.showDetailsModal }
								/>;
							} )
					}
				</ListTable>
			</Paper>
		);
	}
}

const subscriptionProps = PropTypes.oneOfType( [
	PropTypes.arrayOf(
		PropTypes.shape(
			{
				id: PropTypes.string.isRequired,
				icon: PropTypes.string.isRequired,
				name: PropTypes.string.isRequired,
				used: PropTypes.number.isRequired,
				limit: PropTypes.number.isRequired,
				hasNextPayment: PropTypes.bool.isRequired,
				nextPayment: PropTypes.instanceOf( Date ).isRequired,
				hasEndDate: PropTypes.bool.isRequired,
				endDate: PropTypes.instanceOf( Date ).isRequired,
				billingAmount: PropTypes.number.isRequired,
				billingCurrency: PropTypes.string.isRequired,
				status: PropTypes.string.isRequired,
				renewalUrl: PropTypes.string.isRequired,
			}
		)
	),
	PropTypes.object,
] );


Subscriptions.propTypes = {
	subscriptions: subscriptionProps,
	onManage: PropTypes.func.isRequired,
	isGrouped: PropTypes.bool,
	needsAttention: PropTypes.bool,
	showDetailsModal: PropTypes.func,
};

Subscriptions.defaultProps = {
	isGrouped: false,
	needsAttention: false,
	showDetailsModal: null,
};

export default injectIntl( Subscriptions );
