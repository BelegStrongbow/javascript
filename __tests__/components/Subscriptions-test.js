import React from 'react';
import { createComponentWithIntl } from "../../utils";
import Subscriptions from '../../src/components/Subscriptions';
import { MemoryRouter } from "react-router-dom";

test('The Subscriptions component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<MemoryRouter>
			<Subscriptions activeSubscriptions={ [
				{
					id: "3",
					icon: "https://yoast-mercury.s3.amazonaws.com/uploads/2013/02/Yoast_Icon_Large_RGB.png",
					name: "SEO Premium for WordPress",
					used: 4,
					limit: 20,
					nextPayment: new Date("October 13, 2014"),
					billingAmount: 12200,
					billingCurrency: "USD",
					status: "active",
				},
				{
					id: "4",
					icon: "https://yoast-mercury.s3.amazonaws.com/uploads/2013/02/Yoast_Icon_Large_RGB.png",
					name: "SEO Premium for WordPress",
					used: 3,
					limit: 20,
					nextPayment: new Date("October 13, 2014"),
					billingAmount: 12512,
					billingCurrency: "USD",
					status: "active",
				},
			] } onManage={ () => { } } />
		</MemoryRouter>
			);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
