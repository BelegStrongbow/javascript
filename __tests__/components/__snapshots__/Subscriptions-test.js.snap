exports[`test The Subscriptions component matches the snapshot 1`] = `
<ul
  className="jtXnzZ">
  <div
    className="dnTGWH">
    <span>
      <li
        className="gSUJps">
        <span
          className="cpfBRH"
          label="">
          <img
            alt=""
            className="bjVGHc"
            src="https://yoast-mercury.s3.amazonaws.com/uploads/2013/02/Yoast_Icon_Large_RGB.png" />
        </span>
        <span
          className="hSVqo"
          label="Product">
          SEO Premium for WordPress
        </span>
        <span
          className="gWsdrk"
          label="Level">
          20 sites
        </span>
        <span
          className="hHZLIx"
          label="Usage">
          4
          /
          20
        </span>
        <span
          className="jWNjyX"
          label="Next billing on">
          <span>
            10/13/2014
          </span>
        </span>
        <span
          className="Vxkll"
          label="Amount">
          $
           
          125.12
        </span>
        <span
          className="iTexxv"
          label="">
          <a
            href="/subscriptions/3"
            onClick={[Function]} />
        </span>
      </li>
      <li
        className="fTADJm">
        <span
          className="cpfBRH"
          label="">
          <img
            alt=""
            className="bjVGHc"
            src="https://yoast-mercury.s3.amazonaws.com/uploads/2013/02/Yoast_Icon_Large_RGB.png" />
        </span>
        <span
          className="hSVqo"
          label="Product">
          SEO Premium for WordPress
        </span>
        <span
          className="gWsdrk"
          label="Level">
          20 sites
        </span>
        <span
          className="hHZLIx"
          label="Usage">
          3
          /
          20
        </span>
        <span
          className="jWNjyX"
          label="Next billing on">
          <span>
            10/13/2014
          </span>
        </span>
        <span
          className="Vxkll"
          label="Amount">
          $
           
          125.12
        </span>
        <span
          className="iTexxv"
          label="">
          <a
            href="/subscriptions/4"
            onClick={[Function]} />
        </span>
      </li>
    </span>
  </div>
</ul>
`;
